cdap-event-writer-ext-gcp-pubsub/src/main/java/io/cdap/cdap/events/process/pubsub/PubSubEventWriter.java

```diff
@@ -106,6 +103,19 @@ public void initialize(EventWriterContext eventWriterContext) {
     }
   }
 
+  private Publisher.Builder setCredentials(Publisher.Builder publishBuilder, String tokenEndPoint) throws IOException {
+    // Use SA if present
+    if (this.serviceAccountPath != null) {
+      LOG.debug("Using provided service account path for fetching credentials.");
+      return publishBuilder
+        .setCredentialsProvider(
+          () -> GoogleCredentials.fromStream(getCredentials(serviceAccountPath)));
+    }
+
+    // Use token end point for token
+    return publishBuilder.setCredentialsProvider(() -> ComputeEngineCredentials.getOrCreate(tokenEndPoint));
```

Comment:
do we need to check tokenEndPoint not being null?

Reply:
tokenEndPoint is nullable, there is logic to use local credentials in `ComputeEngineCredentials` if it is null .

Reply:
right. I believe it uses LOCAL_COMPUTE_ENGINE_CREDENTIALS. Does the same semantic holds for PubSubEventWriter also?

Reply:
Yes, the same logic applies here.